Source/bmalloc/CMakeLists.txt

```diff
@@ -683,7 +683,9 @@ set_source_files_properties(
     libpas/src/libpas/bmalloc_heap_config.c
     libpas/src/libpas/pas_heap_config_kind.c
     libpas/src/libpas/pas_segregated_page_config_kind.c
-    PROPERTIES LANGUAGE CXX
+    PROPERTIES
+    LANGUAGE CXX
+    COMPILE_OPTIONS "-xc++"
```

Comment:
MSVC isn't currently building bmalloc but if it were then we probably would want to have a `COMPILE_C_AS_CXX` variable that's set based on the compiler and use that.

Reply:
Do you mean adding a `COMPILE_C_AS_CXX` guard around this block so we can turn it on/off from `WebKitCompilerFlags.cmake`?

Reply:
No I mean

```
COMPILE_OPTIONS ${COMPILE_C_AS_CXX}
```